### ISSUE TITLE
OCR plugin not found on NPM registry

### ISSUE THREAD

**Issue by ayan4m1** [nut-tree/nut.js#577]
**Short summary**

[This page](https://nutjs.dev/plugins/ocr) documents a plugin that does [not exist on NPM.js](https://www.npmjs.com/package/@nut-tree/plugin-ocr) - what is going on? Was the plugin pulled from npmjs?

**Desired execution environment / tested on**

- [ ] Virtual machine
- [ ] Docker container
- [x] Dev/Host system

**Node version**: 20.11.0

**OS type and version**:  Win11 x64

**Full code sample related to question**: N/A

**Comment by s1hofmann** [comment]
Hi @ayan4m1 👋 

`@nut-tree/plugin-ocr` is only available via the private nut.js registry, not npm.

You can get access by purchasing a license.

**Comment by ayan4m1** [comment]
You should do a better job updating your documentation so that people do not waste their time like I did. This change to closed source was announced where, exactly?

**Comment by s1hofmann** [comment]
Hi @ayan4m1 👋 

> Not only did you sell out, you also removed all the old versions that were released under an open source license so that others couldn't continue to use out-of-support versions.

`@nut-tree/plugin-ocr` has never been publicly available, nor was any other of the paid addons that were released after introducing non-free addons **almost 2.5 years ago**. 

> You should do a better job updating your documentation so that people do not waste their time like I did. This change to closed source was announced where, exactly? All of your READMEs and documentation sites do not mention this

This is mentioned on the nutjs.dev landing page and e.g. [pricing page](https://nutjs.dev/pricing/pricing). The core readme refers to the website for additional information, so I’d say it’s fair enough to expect people to actually read it.

> tl;dr get off GitHub and npm entirely if you want to do the closed-source thing, kthx.

It was a pleasure to meet you, have a great day!

**Comment by alfredomariamilano** [comment]
> You should do a better job updating your documentation so that people do not waste their time like I did. This change to closed source was announced where, exactly? All of your READMEs and documentation sites do not mention this. Very easy to be confused and very disappointing to me that this went closed-source.
> 
> Not only did you sell out, you also removed all the old versions that were released under an open source license so that others couldn't continue to use out-of-support versions. DISGUSTING.
> 
> _tl;dr_ get off GitHub and npm entirely if you want to do the closed-source thing, kthx.

Wow, you're not just a terrible person for spitting in the face of someone doing free work, but you're also just stupid, since it's written all over the place that the plugin is, in fact, paid. RTFM.

**Comment by reduxionist** [comment]
> Not only did you sell out, you also removed all the old versions that were released under an open source license so that others couldn't continue to use out-of-support versions. DISGUSTING.
> 
> _tl;dr_ get off GitHub and npm entirely if you want to do the closed-source thing, kthx.
@ayan4m1 
You were completely wrong and your attitude is completely unacceptable to professional software developers. If you want to have a career in software, or a reputation in the open-source world, you should publically apologize for falsely slandering and insulting an accomplished developer who contributes to many open-source projects. Poor treatment by users, like that seen this issue, is why so many more maintainers are quitting open-source these days. Please take the mature path and save your own reputation by apologizing for your rush to judgement. One day in the futue you may have maintained an open-source project that many other people use for  some time and then you'll look back at this with regret if you don't apologize for your emotional outburst now...

**Comment by tiritto** [comment]
I just found this place after reading up a post from [I give up](https://nutjs.dev/blog/i-give-up) post.

> You should do a better job updating your documentation so that people do not waste their time like I did. This change to closed source was announced where, exactly? All of your READMEs and documentation sites do not mention this. Very easy to be confused and very disappointing to me that this went closed-source.
> 
> Not only did you sell out, you also removed all the old versions that were released under an open source license so that others couldn't continue to use out-of-support versions. DISGUSTING.
> 
> _tl;dr_ get off GitHub and npm entirely if you want to do the closed-source thing, kthx.

I'm truly baffled. I've been trying to find more respectful wording to describe your behavior, but I just can't find anything that would describe it any gentler, this behavior of yours is simply childish. It shocks me even more given that you have a history of contributing to open-source projects as well, so one would expect you to know better. Especially coming from a person who has full name and professional LinkedIn linked in their profile, rather than being just an anonymous drama seeker.

Nonetheless, it seems you've achieved precisely what you requested. Unfortunately, it has come at the expense of others, tarnishing the experience for everyone. So, thank you for your contribution @ayan4m1 ! You're the best!

**Comment by Summertime** [comment]
> [...] get off GitHub and npm entirely if you want to do the closed-source thing, [...]

GitHub specifically provides guidance on ["closed source" works](https://choosealicense.com/no-permission/) and gives tips on how best to manage them.

https://github.com/npm/npm/issues/8795 Also provides guidance on how NPM supports custom licensing, including "closed source" licenses.

**Comment by ayan4m1** [comment]
> > Not only did you sell out, you also removed all the old versions that were released under an open source license so that others couldn't continue to use out-of-support versions. DISGUSTING.
> > _tl;dr_ get off GitHub and npm entirely if you want to do the closed-source thing, kthx.
> > @ayan4m1
> > You were completely wrong and your attitude is completely unacceptable to professional software developers. If you want to have a career in software, or a reputation in the open-source world, you should publically apologize for falsely slandering and insulting an accomplished developer who contributes to many open-source projects. Poor treatment by users, like that seen this issue, is why so many more maintainers are quitting open-source these days. Please take the mature path and save your own reputation by apologizing for your rush to judgement. One day in the futue you may have maintained an open-source project that many other people use for  some time and then you'll look back at this with regret if you don't apologize for your emotional outburst now...

I'm having a mental health crisis right now. What I said was wrong, I could not see that a few days ago. Take whatever you want from that. I am sorry. Please stop piling on now that I have removed everything. I am seriously ill and need to stop being involved in anything for several months. It was my mistake to post here while I was sick. I am trying to avoid using what people have written here to justify killing myself.
